### ISSUE TITLE
Why message time must be int in MIDI file when it is not in the first place ?

### ISSUE THREAD

**Issue by Atralb** [mido/mido#189]
Hi guys, first of all thanks a lot for your work.

I am beginning to work with MIDI files for deep learning applications and have trouble understanding how to set the time of beginning of a midi message and its duration in the case of this package.

Could someone explain thoroughly what actually is the `time` attribute in a Mido message.

Then even when accepting to not understanding fully the achitecture, when I'm importing an extremely simple midi file create with Musescore, all the messages have a float time attribute. But when reconstructing it after with exactly the same note_on message I get this error. How is it possible ?

Thanks in advance for your answer :)

**Comment by Atralb** [comment]
@olemb ?

**Comment by dbolton** [comment]
I noticed if I iterate through the messages on the file level it gives me floats, but if I iterate at the track level it gives me integers. 

```python

from mido import Message, MidiFile, MidiTrack

mid1 = MidiFile("Violin-1.mid")
for msg in mid1:
    if not msg.is_meta:
        print(msg)
```
time attributes are floats

```python
mid1 = MidiFile("Violin-1.mid")
for msg in mid1.tracks[0]:
    if not msg.is_meta:
        print(msg)
```
time attributes are integers

It is a problem for me too, because I can't save the MIDI messages back if I use the first method (because I get the floats error like you). And I can't save the MIDI messages back if I use the second method (because the resulting file is about 157 times longer/slower when I play it back).

**Comment by dfkettle** [comment]
@dbolton I get this as the conversion factor:

floats = integers / ticks_per_beat

I'm using mido v1.2.9.

**Comment by DanteMarshal** [comment]
> @dbolton I get this as the conversion factor:
> 
> floats = integers / ticks_per_beat
> 
> I'm using mido v1.2.9.

I tried "integer = float * ticks_per_beat" to convert floats to ints, but I get 0.5 values
I think same as @dbolton said, it's "float = integer / (ticks_per_beat * 2)

**Comment by dsheng3** [comment]
Yes, I also have some tempo problems. Its is like @dbolton said. It is now documented https://mido.readthedocs.io/en/latest/midi_files.html#about-the-time-attribute

One is ticks and one is seconds
